packages/sui-consents/src/useUserConsents.js

```diff
@@ -0,0 +1,46 @@
+import {useContext, useEffect, useState} from 'react'
+import SUIContext from '@s-ui/react-context'
+
+import {TCF_WINDOW_API, TCF_VERSION} from './config'
+import hasUserConsents from './hasUserConsents'
+
+/**
+ * @param {array} requiredConsents for example [1, 3, 8]
+ * @return {boolean}
+ */
+export default function useUserConsents(requiredConsents) {
+  /**
+   * Consents acceptance state is inited based on the cookies from the
+   * context, so we know the state of consents from the beginning, even
+   * in SSR.
+   */
+  const {cookies} = useContext(SUIContext)
+  const [areConsentsAccepted, setAreConsentsAccepted] = useState(() =>
+    hasUserConsents({requiredConsents, cookies})
+  )
+
+  /**
+   * From then on, we listen for TCF events so consents changes
+   * can be catched right away, no matter when, and react to them
+   * wherever the hook is used.
+   */
+  useEffect(() => {
```

Comment:
Maybe not a big thing but this is going to add a new event listener for each hook call.

Reply:
That's right. I don't think it's a problem since consents doesn't change a lot while using the app (mostly never). Considering this, I don't if there would be any gain by setting up a context or similar for this. Any other thoughts on this are welcome.

Reply:
Not a big deal, let's keep it in mind in case we see some degradation about this.

Reply:
As @desko27 says consents don't change so often. So imho we could just go ahead and if in a near future if it is a problem we can add the context.